spec/index.html

```diff
@@ -474,331 +612,109 @@
+          for example <code>valueNow</code>, <code>valueMin</code>, and
+          <code>valueMax</code> are of type <code>double</code>, while
+          <code>rowCount</code>, <code>rowIndex</code>, etc. are of type
+          <code>unsigned long</code>.
         </p>
         <p>
-          There are many possible sources of the accessible name
-          of an element from its declarative markup. As an example,
-          an HTML INPUT element may get its accessible name from
-          an associated LABEL element, from the <code>title</code>
-          attribute, from the <code>aria-label</code>
-          attribute, from the <code>aria-labelledby</code>
-          attribute, or more. The [[!ACCNAME-AAM-1.1]] algorithm
-          specifies exactly how these combine to form the final
-          accessible name.
+          The primary difference is with ARIA attributes that take one
+          or more IDREFs. In AccessibleNode, they take references to
+          one or more AccessibleNodes directly. This makes it possible
+          to establish relationships between nodes that aren't in the
+          same <i>tree scope</i>, such as in different frames (with the
+          same origin), or in the shadow DOM of a custom element.
         </p>
         <p>
-          When using the AccessibleNode interface, it's much simpler.
-          The <code>label</code> property determines the accessible
-          name and the <code>description</code> property determines
-          the accessible description. In addition, the
-          <code>labelledBy</code> attribute contains references to
-          other accessible nodes that provided the label for a node,
-          and the <code>describedBy</code> attribute contains
-          references to other accessible nodes that provided
-          the description, but those relation attributes are
-          informational only.
+          In particular,
+          <code>activeDescendant</code>,
+          <code>details</code>,
+          <code>errorMessage</code>, and
+          <code>flowTo</code> are set with another AccessibleNode,
+          while <code>controls</code>, <code>labeledBy</code>, and
+          <code>describedBy</code> all take any array-like object
+          of AccessibleNodes.
         </p>
+      </section>
+      <section id="validation">
+        <h3>
+          Validation
+        </h3>
         <p>
-          In particular, setting an AccessibleNode's
-          <code>labelledBy</code> or <code>describedBy</code>
-          attributes will have <em>no effect</em> on its
-          accessible name or description. The only purpose of
-          setting them is to communicate the relationship between
-          nodes to assistive technology. Set the
-          <code>label</code> and <code>description</code>
-          attributes directly if you want those to change.
+          Upon setting a property of AccessibleNode, no validation is
```

Comment:
Yes, I think that's better for performance. We can always cache it at read
time, but the purpose is to ensure that if an author sets AccessibleNode
properties and doesn't try to read them back and accessibility is never enabled,
we don't waste any time even trying to validate them.